addons/hr_holidays/static/src/views/calendar/year/calendar_year_renderer.js

```diff
@@ -78,6 +78,12 @@ export class TimeOffCalendarYearRenderer extends CalendarYearRenderer {
                 ? classesToAdd.push("o_event_half_left")
                 : classesToAdd.push("o_event_half_right");
         }
+        // handling half pill UX for custom_hours
+        if(record?.rawRecord?.request_unit_hours) {
+            record.start.c.hour < 12
+            ? classesToAdd.push("o_event_half_left")
+            : classesToAdd.push("o_event_half_right");
+        }
```

Comment:
If someone takes time off from 11.30 to 15.30, it will show as if only the morning is taken, no? :thinking: 
I'd do something like this:
```suggestion
        if (record?.rawRecord?.request_unit_hours) {
            if (record.end.c.hour < 12) {
                classesToAdd.push("o_event_half_left")
            } else if (record.end.c.hour > 12) {
                classesToAdd.push("o_event_half_right")
            }
        }
```